Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -770,180 +775,252 @@ section BaseChange
+  map_simp
+
 lemma map_addX (x₁ x₂ L : R) :
-    (W.map φ).toAffine.addX (φ x₁) (φ x₂) (φ L) = φ (W.addX x₁ x₂ L) := by
+    (W.map f).toAffine.addX (f x₁) (f x₂) (f L) = f (W.addX x₁ x₂ L) := by
   simp only [addX]
   map_simp
 set_option linter.uppercaseLean3 false in
 #align weierstrass_curve.base_change_add_X WeierstrassCurve.Affine.map_addX
 
-lemma map_addY' (x₁ x₂ y₁ L : R) :
-    (W.map φ).toAffine.addY' (φ x₁) (φ x₂) (φ y₁) (φ L) = φ (W.addY' x₁ x₂ y₁ L) := by
-  simp only [addY', map_addX]
+lemma map_negAddY (x₁ x₂ y₁ L : R) :
+    (W.map f).toAffine.negAddY (f x₁) (f x₂) (f y₁) (f L) = f (W.negAddY x₁ x₂ y₁ L) := by
+  simp only [negAddY, map_addX]
   map_simp
 set_option linter.uppercaseLean3 false in
-#align weierstrass_curve.base_change_add_Y' WeierstrassCurve.Affine.map_addY'
+#align weierstrass_curve.base_change_add_Y' WeierstrassCurve.Affine.map_negAddY
 
 lemma map_addY (x₁ x₂ y₁ L : R) :
-    (W.map φ).toAffine.addY (φ x₁) (φ x₂) (φ y₁) (φ L) = φ (W.toAffine.addY x₁ x₂ y₁ L) := by
-  simp only [addY, map_addY', map_addX, map_negY]
+    (W.map f).toAffine.addY (f x₁) (f x₂) (f y₁) (f L) = f (W.toAffine.addY x₁ x₂ y₁ L) := by
+  simp only [addY, map_negAddY, map_addX, map_negY]
 set_option linter.uppercaseLean3 false in
 #align weierstrass_curve.base_change_add_Y WeierstrassCurve.Affine.map_addY
 
-lemma map_slope {F : Type u} [Field F] (W : Affine F) {K : Type v} [Field K] (φ : F →+* K)
-    (x₁ x₂ y₁ y₂ : F) : (W.map φ).toAffine.slope (φ x₁) (φ x₂) (φ y₁) (φ y₂) =
-      φ (W.slope x₁ x₂ y₁ y₂) := by
+lemma map_slope {F : Type u} [Field F] (W : Affine F) {K : Type v} [Field K] (f : F →+* K)
+    (x₁ x₂ y₁ y₂ : F) : (W.map f).toAffine.slope (f x₁) (f x₂) (f y₁) (f y₂) =
+      f (W.slope x₁ x₂ y₁ y₂) := by
   by_cases hx : x₁ = x₂
   · by_cases hy : y₁ = W.negY x₂ y₂
-    · rw [slope_of_Yeq hx hy, slope_of_Yeq <| congr_arg _ hx, map_zero]
-      rw [hy, map_negY]
-    · rw [slope_of_Yne hx hy, slope_of_Yne <| congr_arg _ hx]
-      · simp only [negY]
-        map_simp
-      · rw [map_negY]
-        contrapose! hy
-        exact φ.injective hy
-  · rw [slope_of_Xne hx, slope_of_Xne]
-    · map_simp
-    · contrapose! hx
-      exact φ.injective hx
+    · rw [slope_of_Y_eq (congr_arg f hx) <| by rw [hy, map_negY], slope_of_Y_eq hx hy, map_zero]
+    · rw [slope_of_Y_ne (congr_arg f hx) <| W.map_negY f x₂ y₂ ▸ fun h => hy <| f.injective h,
+        map_negY, slope_of_Y_ne hx hy]
+      map_simp
+  · rw [slope_of_X_ne fun h => hx <| f.injective h, slope_of_X_ne hx]
+    map_simp
 #align weierstrass_curve.base_change_slope WeierstrassCurve.Affine.map_slope
 
 variable {R : Type r} [CommRing R] (W : Affine R) {S : Type s} [CommRing S] [Algebra R S]
   {A : Type u} [CommRing A] [Algebra R A] [Algebra S A] [IsScalarTower R S A]
-  {B : Type v} [CommRing B] [Algebra R B] [Algebra S B] [IsScalarTower R S B] (ψ : A →ₐ[S] B)
+  {B : Type v} [CommRing B] [Algebra R B] [Algebra S B] [IsScalarTower R S B] (φ : A →ₐ[S] B)
```

Comment:
Still using φ here?

Reply:
Oops.

Reply:
Seems like I can't reuse the variable `f`, because otherwise I can't make it implicit when I want it to be, thanks to the bug!